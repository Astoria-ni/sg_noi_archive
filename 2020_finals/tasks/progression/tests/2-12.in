25 1000
8 6 4 2 0 -2 -4 -6 -8 -10 0 -4 7 7 7 7 7 7 7 7 7 7 -9 -9 -9
3 1 25
3 6 19
3 8 12
2 24 24 -7 -3
3 17 24
2 12 12 10 -3
1 13 20 2 -5
1 24 24 9 2
1 14 21 -6 -7
3 25 25
2 18 19 -3 0
1 7 14 -9 8
3 5 24
1 20 23 5 -6
1 16 16 2 -8
1 5 8 -7 10
2 1 3 4 2
1 17 22 6 2
1 9 14 -6 -2
2 9 14 2 -5
1 24 25 8 -6
2 6 22 4 -9
3 11 18
3 2 5
3 23 23
1 3 14 -4 -2
2 11 13 -3 -6
3 6 14
3 23 23
3 23 25
2 6 20 -6 -5
2 19 24 -1 9
2 12 13 -3 7
2 14 15 -9 -9
1 7 22 -1 -7
3 2 8
3 8 16
3 14 21
2 20 23 5 -1
3 19 23
2 15 15 -8 7
1 12 18 5 -4
1 6 20 -4 5
2 24 25 8 7
3 21 21
1 1 25 5 -9
3 3 11
3 17 17
1 10 13 9 10
1 10 17 -6 1
2 3 7 -7 -1
2 15 16 2 -10
2 1 4 -9 -6
3 2 10
2 14 21 2 4
2 3 4 9 -9
1 24 24 7 -7
2 15 18 6 9
2 4 11 3 2
2 8 11 -3 -5
1 25 25 3 -3
3 1 16
1 17 23 3 2
2 8 24 7 -10
1 10 11 -2 -10
2 21 23 7 -4
3 14 21
1 25 25 6 -2
3 24 24
3 5 9
2 9 9 -4 2
3 20 23
2 5 10 -9 -7
1 16 22 9 -10
3 25 25
1 12 23 6 -5
1 22 22 2 8
3 20 24
2 15 18 -3 -2
1 24 24 -7 10
2 15 17 -7 -1
2 14 17 -1 10
1 1 9 0 2
3 15 19
3 9 15
2 2 16 0 -1
2 20 24 10 -6
2 24 24 -6 9
3 11 19
3 5 23
1 16 16 -8 -8
2 23 25 -3 0
2 8 11 1 -6
1 10 16 5 7
2 9 15 9 -3
1 18 19 -5 -5
3 18 25
1 21 24 8 -5
3 11 12
2 13 16 -5 1
1 18 21 -2 6
1 5 9 2 -3
1 17 17 3 1
2 24 24 10 7
3 2 7
3 16 21
2 17 19 6 -7
2 14 23 7 3
1 22 23 3 2
3 12 25
2 1 5 2 10
1 19 19 3 -9
2 17 25 9 10
1 6 25 -3 8
2 11 21 -4 -1
3 7 7
2 7 15 -10 -3
1 19 24 2 -4
2 9 25 0 -7
3 8 15
3 22 24
3 22 23
2 7 14 1 2
2 4 15 7 -9
3 12 22
3 6 23
3 4 23
2 9 21 -4 0
1 7 10 -1 -9
3 3 5
3 5 11
2 24 25 -10 0
1 11 11 5 -9
1 20 21 9 5
1 10 15 6 -9
1 5 10 -7 -10
1 20 21 6 3
1 23 25 -1 -9
2 15 25 3 1
2 25 25 10 2
2 20 22 6 -9
3 1 17
2 22 23 -3 -9
3 22 23
2 7 21 7 -10
1 12 16 4 0
1 12 16 5 -1
2 23 24 8 -8
3 12 14
3 2 19
1 1 12 3 2
2 4 4 8 -3
3 25 25
3 2 3
3 21 23
1 18 20 -4 0
2 20 23 -8 1
3 5 18
3 11 25
3 8 18
3 15 18
3 22 24
1 1 5 4 2
1 24 25 -10 -6
2 2 16 1 6
3 2 2
3 15 21
2 13 23 6 -6
1 18 19 -10 9
3 18 24
1 2 9 -4 5
2 18 25 -5 -9
2 19 19 2 -8
1 14 15 8 -7
2 6 7 1 8
3 23 24
2 5 20 -10 4
1 11 12 5 7
3 5 10
2 18 22 2 10
3 22 23
3 9 19
3 2 17
1 11 17 -2 2
3 18 19
2 24 25 -1 4
2 14 23 -10 10
3 1 23
2 14 25 6 6
3 6 6
1 4 14 -2 4
3 7 8
1 9 25 -1 8
1 4 11 7 0
1 5 15 -9 -9
3 22 22
3 5 19
2 14 19 -5 9
3 10 25
1 8 9 -1 -1
2 23 23 -8 10
2 11 20 -5 -8
3 7 12
2 21 21 -4 -7
3 6 7
3 2 4
3 1 23
2 14 21 6 4
3 25 25
1 25 25 1 -4
2 12 13 -9 -2
3 15 24
1 22 24 -8 -1
3 25 25
3 19 25
2 11 14 3 -5
3 22 24
1 11 21 -3 10
3 21 24
3 25 25
1 3 25 8 5
1 22 22 -6 -6
1 4 8 9 -3
1 19 21 6 2
1 4 5 2 9
3 4 20
3 23 24
3 4 12
2 3 13 -4 6
3 3 21
1 14 19 -4 -8
2 1 4 5 2
3 3 9
3 18 20
2 13 19 8 6
1 13 24 7 -4
3 24 24
3 10 19
3 25 25
1 21 21 -5 2
1 17 23 -7 2
1 7 16 3 2
2 16 16 -7 2
3 7 21
1 4 5 10 -5
2 3 25 -10 -2
3 14 23
1 15 17 8 1
1 10 23 6 9
1 13 24 -2 5
1 20 22 -4 -6
2 19 20 -1 -5
2 24 25 10 -3
1 14 14 1 -1
3 8 10
2 21 22 -1 -10
2 3 6 -1 -7
2 8 25 -8 10
1 3 14 -4 4
2 16 19 4 -4
3 8 24
2 22 25 6 10
2 21 22 -9 -8
3 10 23
1 19 20 5 6
3 21 23
1 13 14 -7 -8
3 17 22
2 19 23 4 -5
2 18 20 6 2
2 22 25 -6 -4
2 14 20 5 -9
3 23 25
2 15 20 7 6
2 11 24 -9 -4
2 7 13 -10 5
2 22 24 -9 -1
3 17 20
3 16 23
1 11 24 -2 -1
2 12 16 -8 0
2 9 20 0 -4
1 18 21 9 -10
1 21 25 -6 10
2 19 24 9 -7
1 17 22 7 -2
3 14 19
3 8 19
1 1 25 -7 -1
3 16 16
1 9 17 -7 4
3 22 23
1 21 23 -8 2
3 17 17
3 4 23
1 1 23 -3 8
1 25 25 1 5
3 3 8
1 16 21 3 7
3 4 9
1 13 19 -7 4
2 11 21 8 7
3 22 22
2 5 5 6 9
2 4 15 -9 0
2 23 23 -2 -5
1 22 24 -8 -4
2 23 25 0 -2
1 25 25 1 -10
2 9 9 -1 -9
1 19 19 0 -7
1 15 16 -10 -1
1 14 19 4 4
2 1 1 7 8
1 24 25 -3 2
1 17 18 6 -4
2 14 24 -4 4
1 1 15 5 -10
3 13 23
1 18 24 9 0
1 17 23 5 -2
3 5 7
3 14 20
3 3 14
3 15 20
2 18 20 -6 -9
1 14 23 -4 7
3 23 24
3 11 15
2 19 23 7 -5
2 8 16 -7 -10
2 18 21 5 10
1 25 25 9 2
3 11 18
2 14 19 -4 -3
2 15 15 -4 -2
3 8 13
3 22 24
2 17 17 -7 0
3 11 22
1 19 21 -2 0
2 21 21 -6 -2
2 13 21 -2 9
2 18 23 -5 -4
2 1 6 9 -4
1 25 25 -4 5
2 4 17 10 4
1 3 21 -10 9
2 16 16 -4 0
1 13 14 1 -9
1 25 25 0 -10
3 2 14
2 19 21 -3 4
1 14 17 -5 1
3 20 24
3 4 24
2 19 25 7 -7
1 6 14 -8 9
3 1 4
2 22 23 8 10
3 8 15
3 25 25
3 11 13
1 22 24 5 6
3 8 17
1 25 25 6 9
1 13 14 2 -4
3 5 22
3 17 17
1 10 16 4 -2
2 4 10 -1 4
3 15 25
1 13 13 3 2
3 9 13
3 4 9
2 10 24 -7 -1
3 6 18
2 19 25 -6 7
2 25 25 5 -7
2 15 18 -2 -10
1 13 18 7 5
2 9 23 2 10
1 9 11 4 -9
3 16 18
1 11 23 9 -6
1 10 18 3 10
2 17 22 -4 10
1 23 25 8 7
2 14 21 3 8
3 20 22
1 5 7 -7 1
3 15 25
1 13 14 -9 9
2 17 17 -1 3
3 10 24
1 16 25 1 1
2 11 17 6 -7
1 9 24 2 8
2 3 14 1 -8
1 20 21 4 5
2 5 20 -3 4
3 5 12
1 19 21 3 -8
3 23 23
2 10 17 7 6
2 4 14 6 -7
2 9 25 -7 -5
3 13 21
1 13 24 -2 10
3 10 13
1 25 25 5 8
3 9 19
1 5 18 0 3
1 18 25 -3 -3
3 8 24
3 10 12
1 19 19 -4 -10
3 25 25
3 3 7
1 11 21 6 -8
2 19 20 0 7
2 9 9 10 -10
1 7 20 -1 -5
1 8 23 -9 7
3 25 25
3 18 22
3 21 23
2 15 20 3 3
2 11 21 6 -3
2 13 19 2 1
3 11 14
3 11 13
3 13 13
1 4 15 -3 -7
3 14 25
1 16 18 -10 6
1 7 12 9 -6
2 14 20 7 -4
1 14 23 -4 10
1 19 23 -3 6
1 17 23 7 -9
2 10 19 -7 0
3 19 23
1 12 25 9 9
2 15 16 -4 3
3 25 25
2 5 13 2 -4
3 19 20
2 3 23 -3 8
3 15 24
2 2 7 4 -4
1 14 16 8 2
3 12 15
3 4 17
2 18 20 -10 -10
3 22 24
1 3 21 0 10
2 7 7 6 9
1 13 23 -8 -3
2 4 24 -8 -7
2 10 25 5 -10
1 9 23 8 -5
3 16 19
2 3 6 3 -7
2 5 15 -4 -10
3 2 6
2 24 25 -6 -10
3 3 20
3 7 15
1 8 8 -3 -2
3 22 24
3 3 18
2 4 21 7 -10
2 4 7 -5 0
3 24 24
1 6 21 -8 -9
3 8 21
1 17 23 -9 9
1 10 17 -5 0
3 20 21
3 19 20
2 23 23 3 -2
1 22 25 7 5
3 19 22
2 10 17 10 0
2 21 25 5 2
2 17 22 5 2
2 11 12 -8 -10
3 11 18
3 13 14
1 9 9 9 2
1 19 22 8 10
2 18 21 -3 1
2 10 12 7 -9
3 3 4
2 15 16 1 2
3 16 18
2 3 13 -8 8
1 6 20 -6 5
2 9 21 -8 8
1 17 20 5 8
3 17 18
1 18 23 2 2
2 19 19 -6 -8
1 7 8 10 -4
3 18 25
3 8 8
1 9 18 -7 -4
2 22 23 6 2
1 11 15 9 -1
1 5 18 9 6
2 8 20 -5 -6
3 11 19
1 8 12 -1 -1
2 13 18 4 8
3 24 25
3 3 23
1 23 23 -1 -3
3 24 25
3 6 15
3 3 21
2 20 22 6 6
2 5 15 -2 -5
3 25 25
1 21 24 2 -8
1 7 18 6 -9
2 4 5 -2 8
1 21 21 -2 -10
1 3 12 -6 -6
3 25 25
3 6 24
1 14 15 -9 9
2 3 18 7 -2
2 14 19 6 3
2 8 20 -2 -9
2 4 12 -1 -2
1 10 17 -1 10
3 8 15
1 19 25 4 -5
1 4 4 8 3
2 18 25 -5 -8
1 11 15 2 -5
3 17 25
3 9 16
2 10 17 4 4
2 11 17 7 0
1 21 24 4 -6
3 3 17
3 20 22
3 25 25
3 5 22
3 16 24
3 9 18
1 1 17 5 8
2 7 20 4 -6
1 10 18 4 5
2 21 22 4 -4
3 14 19
1 4 7 8 9
1 10 17 10 -5
1 11 16 -5 -5
1 20 24 3 5
2 24 24 -7 -2
3 2 10
2 23 25 -4 8
2 9 10 -4 8
1 17 19 4 0
2 14 22 0 7
1 18 24 -7 6
1 17 18 4 -10
2 21 24 -8 -2
1 6 7 -3 -2
1 11 13 4 -2
2 25 25 10 2
2 24 24 10 9
2 4 10 -10 9
1 20 23 8 6
2 10 16 5 10
1 6 21 2 4
1 21 24 1 -1
2 6 21 -1 -3
1 1 14 0 6
3 3 7
1 9 25 5 -6
3 17 17
2 11 21 -3 3
1 25 25 -7 -8
3 2 2
3 22 24
1 16 18 6 -7
1 13 24 -3 -7
1 4 23 9 7
3 8 17
3 13 24
1 3 3 9 7
3 18 20
3 3 7
3 4 20
2 14 20 -7 -9
3 25 25
2 8 16 -10 8
2 11 16 -6 1
2 21 25 -1 -10
1 14 17 -7 7
2 2 2 4 10
1 1 13 -8 9
3 23 24
2 6 22 3 0
1 25 25 7 -8
3 2 13
1 14 19 0 4
3 23 25
3 22 22
3 24 25
1 3 19 1 -2
2 15 19 0 0
3 9 9
1 5 15 -2 -8
2 25 25 0 4
2 3 22 3 7
1 21 25 -5 -6
2 10 15 -1 4
2 10 22 -9 -4
2 22 25 -7 -5
1 21 25 -3 -3
3 13 14
1 16 25 7 5
1 16 21 4 -10
1 11 13 -10 2
2 14 24 -5 -5
2 1 24 -1 -8
3 17 25
2 2 14 -9 -2
2 6 16 2 8
3 9 17
1 9 20 3 5
1 6 13 2 6
3 5 22
1 15 22 -3 9
1 13 20 8 -6
2 15 21 -9 8
1 10 18 9 2
2 10 21 1 0
2 3 23 -7 8
1 14 14 10 -6
1 7 8 -6 2
1 1 2 -4 1
2 21 21 4 -8
1 11 14 1 -2
3 15 23
1 1 25 8 10
2 17 25 -9 3
1 20 22 7 7
2 6 12 4 -4
2 20 20 9 2
2 17 25 -1 10
3 3 6
2 3 16 -3 4
1 1 11 -7 10
1 15 23 -9 -10
2 1 7 7 -2
2 18 23 10 0
3 10 14
1 9 11 -6 -2
2 6 24 0 9
3 3 8
1 20 23 -1 -8
2 13 15 -7 8
3 14 18
3 13 13
2 7 25 3 -3
2 25 25 10 9
2 12 23 -5 4
2 13 20 9 8
3 25 25
3 17 22
2 5 8 -9 -2
3 1 25
2 18 18 7 -9
1 12 21 4 3
3 18 24
1 11 14 -10 8
1 12 16 2 -10
1 18 22 8 6
2 2 21 -9 10
2 24 24 5 2
3 14 23
3 5 21
2 21 23 7 -8
1 4 25 -10 3
3 20 21
2 23 23 0 3
1 18 22 -9 4
1 12 13 -8 8
3 14 16
3 8 14
1 3 3 8 -10
1 4 20 -7 -2
3 4 12
3 7 22
2 6 15 5 4
3 23 25
1 6 23 -2 7
3 8 8
1 9 10 -6 -4
1 8 23 7 10
1 20 21 -10 4
1 25 25 -4 -9
3 24 24
1 1 16 -5 -5
2 3 23 7 2
2 25 25 -4 5
3 4 4
3 12 13
1 25 25 3 2
3 9 19
1 15 19 -1 -7
3 22 22
2 13 23 -10 9
2 13 17 5 1
2 14 18 0 2
2 9 18 4 -7
3 14 21
3 10 21
2 23 25 2 -10
2 18 24 -6 -8
3 22 23
3 5 5
3 4 5
3 16 20
1 4 12 -3 0
3 1 24
2 5 24 -4 -9
2 16 18 -6 -3
2 3 17 1 -10
2 19 19 -6 4
2 25 25 -8 -9
3 6 10
1 14 25 -4 -6
2 1 2 9 6
2 9 11 -9 -1
3 24 25
1 23 23 0 -5
1 6 18 -2 -4
1 1 13 -6 -9
1 9 21 1 3
2 7 16 -9 1
2 20 24 2 -7
3 18 20
3 18 24
3 12 17
3 19 21
2 4 16 -3 3
2 13 20 6 6
2 21 23 4 -6
3 16 25
2 17 22 2 8
2 5 20 3 10
3 24 25
3 3 23
3 4 19
2 8 15 2 1
3 9 14
3 17 23
1 10 24 1 -10
2 15 21 6 0
1 23 24 0 6
2 16 16 -1 8
3 15 21
3 10 22
3 6 9
3 8 14
3 4 20
2 21 21 5 3
1 2 7 1 -3
2 3 3 -7 6
1 4 10 6 -9
3 3 4
2 7 8 10 -1
2 4 4 2 4
3 14 16
3 8 19
2 9 25 6 0
1 23 24 -6 10
2 5 16 5 1
3 19 25
3 21 25
3 10 21
1 9 25 2 7
1 23 25 4 8
3 23 25
3 11 22
1 2 11 4 5
3 14 19
2 13 17 2 -3
2 7 7 -6 -10
3 19 19
2 12 15 -7 -1
1 15 23 -8 0
2 20 20 3 -9
3 22 22
1 11 23 -10 10
2 16 18 -1 -8
1 22 24 10 -8
3 14 19
3 16 19
3 12 17
1 18 21 -5 -3
1 16 16 8 6
3 25 25
1 24 25 5 10
3 8 10
2 24 25 -8 7
1 14 16 5 -6
1 8 15 8 9
2 8 8 7 6
3 14 16
1 3 14 -10 -7
3 25 25
1 19 19 10 2
1 9 21 -10 5
2 19 24 -1 3
2 8 10 1 6
1 8 16 -10 4
1 13 13 -8 1
2 16 19 4 -1
3 22 24
2 15 25 -10 -6
2 3 16 -7 -1
1 14 19 0 5
2 15 17 1 -7
3 12 23
3 21 24
3 14 25
1 2 5 -7 -4
2 22 22 -3 9
2 21 22 6 -8
1 21 24 -4 5
2 9 11 2 -7
1 10 13 -3 -6
2 13 19 5 5
3 6 16
3 13 15
1 13 16 -1 -9
1 11 19 -5 4
1 4 21 7 -8
2 21 23 10 -1
2 2 5 9 -10
1 3 20 -7 -7
3 21 23
2 6 24 2 8
2 24 24 5 -10
1 5 22 5 9
2 9 21 5 -4
2 21 23 -1 9
1 13 14 1 6
2 18 20 3 -1
3 17 23
3 23 24
2 13 24 -3 -9
2 25 25 -1 6
1 13 20 -2 -9
1 3 23 -4 -8
2 1 25 6 10
1 21 25 -10 8
2 6 8 -3 -10
3 17 21
1 25 25 9 3
2 24 25 -7 9
1 1 21 -5 9
2 6 7 4 -3
3 11 21
3 2 19
3 7 12
2 16 20 -2 1
3 11 14
1 1 7 9 5
1 16 19 -8 2
3 11 25
2 14 21 -5 -5
3 5 7
3 21 23
2 11 19 -6 1
2 17 22 -4 -7
1 18 21 3 10
3 1 10
3 25 25
2 13 16 9 5
3 5 12
1 1 19 -6 8
1 2 22 -10 -8
1 11 25 3 4
1 17 22 10 9
2 12 25 2 7
2 17 24 5 -8
3 8 8
2 12 19 7 -8
3 18 21
2 15 23 5 9
2 17 20 8 3
1 3 11 7 -9
1 1 1 3 5
3 24 24
3 1 25
1 12 14 -6 10
3 22 23
1 5 23 -9 -2
2 19 24 8 -4
3 5 5
1 21 21 -3 2
3 13 17
3 11 16
3 19 21
2 1 8 -7 -9
2 3 11 10 3
3 10 14
3 19 25
3 21 21
3 22 25
3 9 18
2 1 22 -5 -9
3 25 25
1 5 5 9 -8
1 14 25 -5 8
3 2 8
1 5 25 -2 7
1 21 21 -6 0
1 6 9 1 4
2 2 8 -9 9
3 5 23
3 22 23
1 6 18 4 -7
1 18 24 9 -6
1 22 22 1 -7
1 18 25 3 -4
1 21 24 1 -3
2 5 19 -4 4
2 1 24 8 4
3 25 25
3 17 23
1 8 8 7 -5
2 6 19 -7 -9
2 7 18 -10 8
3 22 22
3 23 23
2 19 19 3 -1
3 19 22
2 23 24 5 -9
2 8 21 -1 6
1 24 24 -8 7
3 8 17
2 20 25 -9 3
2 16 19 10 -7
3 13 21
2 22 22 3 1
1 8 22 -1 -6
3 9 19
1 14 18 3 1
3 8 14
2 4 14 -10 -6
3 18 23
3 8 25
3 12 15
2 3 10 -9 10
2 7 19 4 0
3 1 17
2 15 20 -9 7
1 12 25 3 -2
3 3 7
1 18 23 10 -3
2 14 24 7 -7
3 7 13
2 4 21 7 -6
1 11 18 -10 2
2 9 20 -9 8
1 4 24 -6 4
3 4 21
1 8 8 1 -9
2 25 25 8 9
1 22 25 1 0
3 14 22
1 11 25 6 -3
1 10 14 -10 -6
3 13 18
3 23 24
2 25 25 -2 -4
3 13 24
2 6 15 -4 3
3 1 20
2 22 25 4 -9
2 5 16 8 9
1 4 5 -10 -10
1 22 23 10 3
2 23 24 5 4
1 1 18 6 -5
3 17 24
2 25 25 -4 8
1 1 22 6 0
3 9 24
3 1 25
