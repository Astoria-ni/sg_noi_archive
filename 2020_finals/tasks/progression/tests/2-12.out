10
7
3
6
1
4
8
2
1
5
1
2
2
4
3
4
1
5
1
3
4
7
1
3
3
1
2
3
2
6
12
3
2
3
2
8
1
7
3
2
7
10
12
2
2
3
2
3
5
1
2
2
5
4
4
2
2
1
2
4
2
6
2
5
6
2
10
1
2
1
4
6
2
2
2
10
1
7
1
3
2
2
1
4
2
2
11
5
2
1
7
1
6
10
2
7
5
2
2
3
4
6
3
8
1
2
1
8
4
3
1
8
3
3
5
3
2
3
6
6
2
3
9
2
9
2
5
1
2
5
7
1
3
2
6
9
3
2
7
4
8
1
9
3
7
6
3
1
3
1
2
2
2
2
1
4
2
1
2
9
2
6
2
4
2
11
9
2
7
1
14
2
2
3
4
2
2
2
2
2
1
9
2
6
2
5
6
1
1
6
3
8
3
7
2
1
7
4
7
5
2
2
1
1
2
3
3
2
2
3
1
2
8
2
1
2
1
2
8
8
5
6
3
3
3
3
1
1
4
8
2
8
17
2
3
4
8
5
2
1
1
1
2
11
1
5
9
2
1
2
3
7
5
2
2
5
5
2
5
2
16
15
6
4
5
5
2
5
5
2
2
4
4
2
7
2
6
3
1
1
3
3
3
1
2
2
1
3
4
4
6
4
3
3
3
2
4
10
11
4
4
5
2
2
5
1
2
1
2
1
4
2
1
2
3
2
3
3
1
3
3
1
3
7
2
1
7
1
1
3
10
4
7
6
2
7
3
11
4
5
13
7
4
2
6
10
2
8
11
